src/app/gui/map/mapservice.js

```diff
@@ -176,18 +176,24 @@ class OlMapViewer {
    * @FIXME add description
    */
   goToRes(coordinates, options = {}) {
-    const view       = this.map.getView();
-    const animate    = options.animate || true;
-    const resolution = options.resolution || view.getResolution();
-    if (animate) {
-      view.animate(
-        { duration: 200, center: coordinates },
-        { duration: 200, resolution }
-      );
-    } else {
-      view.setCenter(coordinates);
-      view.setResolution(resolution);
-    }
+    return new Promise((resolve) => {
+      const view       = this.map.getView();
+      const animate    = options.animate || true;
+      const resolution = options.resolution || view.getResolution();
+      const key = view.on('change:center', () => {
+        ol.Observable.unByKey(key);
+        setTimeout(resolve);
+      });
+      if (animate) {
+        view.animate(
+          { duration: 200, center: coordinates },
+          { duration: 200, resolution }
+        );
+      } else {
+        view.setCenter(coordinates);
+        view.setResolution(resolution);
+      }
+    })
```

Comment:
@volterra79 the else condition will never be reached (ref: `options.animate || true`)

Reply:
@Raruto right. Need to change it. Need to check if boolean otherwise set true